src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.Cache.cs

```diff
@@ -582,18 +587,34 @@ internal void InitializePropCache()
             }
 
             JsonPropertyInfo[] array = PropInitFunc(context);
-            var properties = new JsonPropertyDictionary<JsonPropertyInfo>(Options.PropertyNameCaseInsensitive, array.Length);
+            Dictionary<string, JsonPropertyInfo>? ignoredMembers = null;
+            JsonPropertyDictionary<JsonPropertyInfo> propertyCache = new(Options.PropertyNameCaseInsensitive, array.Length);
+
             for (int i = 0; i < array.Length; i++)
             {
-                JsonPropertyInfo property = array[i];
-                if (!properties.TryAdd(property.NameAsString, property))
+                JsonPropertyInfo jsonPropertyInfo = array[i];
+                bool hasJsonInclude = jsonPropertyInfo.SrcGen_HasJsonInclude;
```

Comment:
Move `hasJsonInclude` to the `if` below? (not used elsewhere)

Reply:
Hmm it's used on L608 below (where we skip fields unless opted-into by users).